web/html/src/manager/minion/packages/package-states.js

```diff
@@ -1,399 +1,357 @@
-                <td colSpan="2">
-                    <div>{t("No package states.")}</div>
-                </td>
-            </tr>
-        }
+      {elements.length > 0 ? elements :
+        <tr>
+          <td colSpan="2">
+            {
+              view === "changes"
+                ? t("No current changes.")
+                : t("No package states.")
+            }
+          </td>
+        </tr>
+      }
       </tbody>
     );
-  }
-
-  onSearchChange(event) {
-    this.setState({
-        filter: event.target.value
-    });
-  }
-
-  handleUndo(packageState) {
-      return event => {
-         this.state.changed.delete(packageStateKey(packageState));
-         this.setState({
-            changed: this.state.changed
-         });
-      }
-  }
-
-  addChanged(original, newPackageStateId, newVersionConstraintId) {
-      const key = packageStateKey(original);
-      const currentState = this.state.changed.get(key);
-      if (currentState != undefined &&
-          newPackageStateId ==  currentState.original.packageStateId &&
-          newVersionConstraintId ==  currentState.original.versionConstraintId) {
-            this.state.changed.delete(key);
-      } else {
-            this.state.changed.set(key, {
-                original: original,
-                value: {
-                    arch: original.arch,
-                    epoch: original.epoch,
-                    version: original.version,
-                    release: original.release,
-                    name: original.name,
-                    packageStateId: newPackageStateId,
-                    versionConstraintId: newVersionConstraintId
-                }
-            });
-      }
-      this.setState({
-         changed: this.state.changed
-      });
-  }
-
-  handleStateChange(original) {
-      return event => {
-         const newPackageStateId = selectValue2PackageState(parseInt(event.target.value));
-         this.addChanged(
-            original,
-            newPackageStateId,
-            newPackageStateId == INSTALLED ? LATEST :  original.versionConstraintId
-         );
-      }
-  }
-
-  handleConstraintChange(original) {
-      return event => {
-         const newPackageConstraintId = selectValue2VersionConstraints(parseInt(event.target.value));
-         const key = packageStateKey(original);
-         const currentState = this.state.changed.get(key);
-         const currentPackageStateId = currentState != undefined ? currentState.value.packageStateId : original.packageStateId;
-         this.addChanged(
-            original,
-            currentPackageStateId,
-            newPackageConstraintId
-         );
-      }
-  }
-
-
-  render() {
+  };
+
+  const renderState = (row, currentState) => {
+    let versionConstraintSelect = null;
+    let undoButton = null;
+
+    if (currentState.packageStateId === packageHelpers.INSTALLED) {
+      versionConstraintSelect =
+        <Select id={currentState.name + "-version-constraint"}
```

Comment:
let's keep the usage of the native <select /> as before. This Select component only makes sense inside a <Form /> wrapper..

Reply:
I addressed your wish in b4f21e7